### PR TITLE
ci: push bin image to Docker Hub

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/moby/moby/pull/45519

**- What I did**

Update bin-image workflow to push the bin image to Docker Hub

image
> https://hub.docker.com/r/crazymax/moby-bin/tags

```
$ undock --rm-dist --all crazymax/moby-bin:latest ./moby-bin
...
$ tree -nh ./moby-bin/
[4.0K]  ./moby-bin/
├── [4.0K]  linux_amd64
│   ├── [ 37M]  containerd
│   ├── [8.1M]  containerd-shim-runc-v2
│   ├── [ 18M]  ctr
│   ├── [748K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 62M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 11M]  rootlesskit
│   ├── [6.9M]  rootlesskit-docker-proxy
│   ├── [ 14M]  runc
│   └── [ 32M]  vpnkit
├── [4.0K]  linux_arm64
│   ├── [ 36M]  containerd
│   ├── [7.9M]  containerd-shim-runc-v2
│   ├── [ 17M]  ctr
│   ├── [527K]  docker-init
│   ├── [2.0M]  docker-proxy
│   ├── [ 59M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.6M]  rootlesskit-docker-proxy
│   ├── [ 13M]  runc
│   └── [ 38M]  vpnkit
├── [4.0K]  linux_armv5
│   ├── [ 35M]  containerd
│   ├── [7.8M]  containerd-shim-runc-v2
│   ├── [ 17M]  ctr
│   ├── [484K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 57M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.8M]  rootlesskit-docker-proxy
│   └── [ 13M]  runc
├── [4.0K]  linux_armv6
│   ├── [ 35M]  containerd
│   ├── [7.8M]  containerd-shim-runc-v2
│   ├── [ 17M]  ctr
│   ├── [484K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 57M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.8M]  rootlesskit-docker-proxy
│   └── [ 13M]  runc
├── [4.0K]  linux_armv7
│   ├── [ 35M]  containerd
│   ├── [7.8M]  containerd-shim-runc-v2
│   ├── [ 17M]  ctr
│   ├── [364K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 57M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.8M]  rootlesskit-docker-proxy
│   └── [ 12M]  runc
├── [4.0K]  linux_ppc64le
│   ├── [ 36M]  containerd
│   ├── [8.0M]  containerd-shim-runc-v2
│   ├── [ 18M]  ctr
│   ├── [843K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 60M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.7M]  rootlesskit-docker-proxy
│   └── [ 13M]  runc
├── [4.0K]  linux_s390x
│   ├── [ 39M]  containerd
│   ├── [8.6M]  containerd-shim-runc-v2
│   ├── [ 19M]  ctr
│   ├── [615K]  docker-init
│   ├── [2.0M]  docker-proxy
│   ├── [ 64M]  dockerd
│   ├── [ 14K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 11M]  rootlesskit
│   ├── [7.1M]  rootlesskit-docker-proxy
│   └── [ 14M]  runc
└── [4.0K]  windows_amd64
    ├── [103K]  containerutility.exe
    ├── [2.1M]  docker-proxy.exe
    └── [ 57M]  dockerd.exe

8 directories, 82 files
```

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

cc @rumpl @vvoland

### CODE REVIEW

File: docker-bake.hcl [hunk 1]
```diff
@@ -138,6 +143,20 @@ target "all-cross" {
   inherits = ["all", "_platforms"]
 }
 
+#
+# bin image
+#
+
+target "bin-image" {
```
Comment (rumpl) [inline docker-bake.hcl:150]: This one is for building locally for the current platform right?

Comment (crazy-max) [reply docker-bake.hcl:150]: Yes

**Review by rumpl — APPROVED** [review]

File: .github/workflows/bin-image.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      -
+        name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_MOBYBIN_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_MOBYBIN_TOKEN }}
```
Comment (crazy-max) [inline .github/workflows/bin-image.yml:71]: Arbitrary secrets, would need to create them on this repo and also create Hub repo `moby/bin`. cc @thaJeztah @sam-thibault

**Review by vvoland — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
> linux_armv5

Do we still support arm v5? 👀

File: .github/workflows/bin-image.yml (hunk 2 above)
Comment (crazy-max) [reply .github/workflows/bin-image.yml:71]: Keeping in draft as we need secrets to be added on this repo

**Comment by crazy-max** [comment]
> > linux_armv5
> 
> Do we still support arm v5? 👀

Removed `arm/v5`

File: .github/workflows/bin-image.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    tags:
+      - 'v*'
+  pull_request:
+
+env:
+  MOBYBIN_REPO_SLUG: moby/moby-bin
+  PLATFORM: Moby Engine
+  PRODUCT: Moby
+  DEFAULT_PRODUCT_LICENSE: Moby
+  PACKAGER_NAME: Moby
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply .github/workflows/bin-image.yml:22]: > Does the bakefile set proper defaults for these (if nothing is passed?)

Nothing is passed atm, hence setting them in the workflow

> We should look at splitting this PR into;

Sure will open another PR with just bake changes

[the remark this answers is not in the document]
Comment (crazy-max) [reply .github/workflows/bin-image.yml:22]: Ah recall we had this convo: https://github.com/moby/moby/pull/43529#discussion_r925254956

**Comment by neersighted** [comment]
Secrets are on the repo; re-opening to trigger CI.

**Review by neersighted — APPROVED** [review]

**Pull request merged**